File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) 
    {
        int sum{};
        int ans{INT_MIN};
        int n=nums.size();
        for(int i{};i<n;i++)
        {
            sum+=nums[i];
            ans=max(sum,ans);
            sum=max(sum,0);
        }
        return ans;
        
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) 
    {
        int n=nums.length;
        int sum=0;
        int ans=Integer.MIN_VALUE;
        for(int i=0;i<n;i++)
        {
            sum+=nums[i];
            ans=Math.max(ans,sum);
            sum=Math.max(sum,0);
        }
        return ans;
    }
}